### PR TITLE
Generate report hash fix

### PR DESCRIPTION
**Pull request by csordasmarton**
Use files from the zip to generate report hash in case of storing run.

### CODE REVIEW

File: libcodechecker/server/api/report_server.py [hunk 1]
```diff
@@ -1987,7 +1987,7 @@ def massStoreRun(self, name, tag, version, b64zip, force):
                     # second time here. Use re-use the
```
Comment (gyorb) [inline libcodechecker/server/api/report_server.py]: Where is the first plist parse? Shouldn't that be updated too? Or the comment is outdated?

Comment (csordasmarton) [reply libcodechecker/server/api/report_server.py]: The first plist parse is at the client side. We don't have to update those too.

File: libcodechecker/analyze/plist_parser.py [hunk 2]
```diff
@@ -102,8 +103,12 @@ def parse_plist(path):
 
             # We need to extend information for plist files generated
             # by older clang version (before 3.8).
+            file_path = files[diag['location']['file']]
+            if source_root is not None:
```
Comment (whisperity) [inline libcodechecker/analyze/plist_parser.py]: `if x is not None` can be replaced with `if x` if `source_root` isn't a boolean.

File: libcodechecker/server/api/report_server.py (hunk 1 above)
Comment (gyorb) [reply libcodechecker/server/api/report_server.py]: In that case this comment is outdated here right? We can not reuse the results from the client side.

**Pull request merged**